crates/bevy_ecs/src/world/mod.rs

```diff
@@ -860,85 +682,244 @@ impl World {
+    }
+
+    /// Retrieves an [`EntityWorldMut`] that exposes read and write operations for the given `entity`.
     ///
     /// # Errors
     ///
-    /// - Returns [`EntityMutableFetchError::EntityDoesNotExist`] if any of the given `entities` do not exist in the world.
-    ///     - Only the first entity found to be missing will be returned.
-    /// - Returns [`EntityMutableFetchError::AliasedMutability`] if the same entity is requested multiple times.
+    /// [`EntityDoesNotExistError`] if any entity does not exist in the world.
     ///
-    /// # Examples
+    /// ```
+    /// use bevy_ecs::{component::Component, world::World};
     ///
-    /// For examples, see [`World::entity_mut`].
+    /// #[derive(Component)]
+    /// struct Position {
+    ///   x: f32,
+    ///   y: f32,
+    /// }
     ///
-    /// [`EntityHashSet`]: crate::entity::hash_set::EntityHashSet
+    /// let mut world = World::new();
+    /// let entity = world.spawn(Position { x: 0.0, y: 0.0 }).id();
+    /// let mut entity_mut = world.get_entity_mut(entity).unwrap();
+    /// let mut position = entity_mut.get_mut::<Position>().unwrap();
+    /// position.x = 1.0;
+    /// ```
     #[inline]
-    pub fn get_entity_mut<F: WorldEntityFetch>(
+    pub fn get_entity_mut(
         &mut self,
-        entities: F,
-    ) -> Result<F::Mut<'_>, EntityMutableFetchError> {
+        entity: Entity,
+    ) -> Result<EntityWorldMut<'_>, EntityMutableFetchError> {
         let cell = self.as_unsafe_world_cell();
-        // SAFETY: `&mut self` gives mutable access to the entire world,
-        // and prevents any other access to the world.
-        unsafe { entities.fetch_mut(cell) }
+        let location = cell
+            .entities()
+            .get(entity)
+            .ok_or(EntityDoesNotExistError::new(entity, cell.entities()))?;
+        // SAFETY: We have mutable access to the entire world
+        let world = unsafe { cell.world_mut() };
+        // SAFETY: location was fetched from the same world's `Entities`.
+        Ok(unsafe { EntityWorldMut::new(world, entity, location) })
+    }
+
+    /// Gets mutable access to multiple entities.
+    ///
+    /// # Errors
+    ///
+    /// - [`EntityMutableFetchError::EntityDoesNotExist`] if any entities do not exist in the world
+    /// - [`EntityMutableFetchError::AliasedMutability`] if the same entity is specified multiple times
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// let mut world = World::new();
+    /// # let id1 = world.spawn_empty().id();
+    /// # let id2 = world.spawn_empty().id();
+    ///
+    /// // Disjoint mutable access.
+    /// let [entity1, entity2] = world.get_many_entities_mut([id1, id2]).unwrap();
+    ///
+    /// // Trying to access the same entity multiple times will fail.
+    /// assert!(world.get_many_entities_mut([id1, id1]).is_err());
+    /// ```
+    pub fn get_many_entities_mut<T: TrustedEntityBorrow, const N: usize>(
```

Comment:
https://github.com/bevyengine/bevy/pull/18315#discussion_r1996461020

> > it looks like I left a few as `T: TrustedEntityBorrow`. Should I switch those back to `Entity`, too?
>
> Did you? I don't see them!

Here's one!  I'll switch `get_many_entities_mut` to `[Entity; N]` and `get_many_entities_unique_mut` to `UniqueEntityArray<N>`.